pkg/controllers/managementuser/rbac/podsecuritypolicy/version.go

```diff
@@ -0,0 +1,31 @@
+package podsecuritypolicy
+
+import (
+	"errors"
+	"fmt"
+	"strconv"
+
+	v3 "github.com/rancher/rancher/pkg/generated/norman/management.cattle.io/v3"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var errVersionIncompatible = errors.New("podsecuritypolicies are not available in Kubernetes v1.25 and above")
+var genericErrorString = "failed to check cluster version compatibility for podsecuritypolicy controllers: %v"
+
+func checkVersion(clusterName string, clusterLister v3.ClusterInterface) error {
+	cluster, err := clusterLister.Get(clusterName, v1.GetOptions{})
+	if err != nil {
+		return fmt.Errorf("failed to get cluster [%s]: %w", clusterName, err)
+	}
+	if cluster.Status.Version == nil {
+		return fmt.Errorf("cannot validate Kubernetes version for podsecuritypolicy capability: cluster [%s] status version is not available yet", clusterName)
+	}
+	v, err := strconv.Atoi(cluster.Status.Version.Minor)
```

Comment:
Might not be critical for now, but do we need to future-proof for v 2.X and limit this check to require v1.X?

Reply:
Good point, I should probably follow other examples of where we check semantic versions, using the `semver` package...